keps/sig-node/5328-node-capabilities/README.md

```diff
@@ -0,0 +1,821 @@
+*   On a mismatch, the plugin will return an FailedScheduling status with a reason that includes the capability that was not found on Nodes.
+
+
+### Admission Controller Changes
+
+To enable the validation of API requests against Node Capabilities, this KEP proposes the introduction of a new admission controller plugin `NodeCapabilityValidator `that will be enabled when the NodeCapabilities feature gate is active. This is necessary because the validation logic that requires cross-referencing objects i,e looking up the `Node` object that a `Pod` is bound to (spec.nodeName). THe admission controller would 
+
+*   Inspects incoming pod update request and infer if it depends on any node capability. 
+*   It verifies that `pod.spec.nodeName` is set. If the pod is not yet bound to a node, the validator takes no action.
+*   Retrieve the `Node` object corresponding to` pod.spec.nodeName` and check the node.Status.capabilities map of the fetched Node object.
+*   If the capability is missing or has a non-matching value, the request is marked as invalid
+
+### Example Walkthrough
+
+**New Features and Version Skew Management**
+
+1. When developing a new Kubelet feature Eg:[ In-Place Pod Resize for Guaranteed QoS pods](https://github.com/kubernetes/enhancements/issues/5294), the skew management strategy involves
+    *   Modifying the Kubelet to advertise the feature as a Node Capability if the feature gate `InPlacePodVerticalScalingExclusiveCPUs` is active. Kubelet would add <code>[kubernetes.io/feature/inPlacePodResizeExclusiveCpus](http://kubernetes.io/feature/inPlacePodResizeExclusiveCpus): true </code>in<code> status.capabilities</code>.
+    *   Updating the <code>NodeCapabilityValidator</code> to check if the node specified in the pod's <code>spec.nodeName</code> possesses this capability before the request is admitted.
+2. An Admin enables the <code>InPlacePodVerticalScalingExclusiveCPUs</code> alpha feature gate on a NodePool. 
+3. The Kubelet on the Node detects this and advertises its support by setting <code>[kubernetes.io/feature/inPlacePodResizeExclusiveCpus](http://kubernetes.io/feature/inPlacePodResizeExclusiveCpus): true </code>in<code> status.capabilities</code>.<code> </code>
+4. An Application Owner has a Guaranteed QoS Pod running on the Node and requests an in place resize. 
+5. When the API server processes the resize request, its validation logic for pod updates is invoked in the admission controller. The request is admitted only if the Node supports the feature. \
+
+
+**Workload Requirements during Scheduling**
+
+1. The feature developer will enhance the Kubelet to conditionally advertise capabilities for `PodLevelResources`. Kubelet will check that both the feature gate is enabled and the underlying OS is Linux before adding the `kubernetes.io/feature/PodLevelResources: "true"` capability to the node's status.
```

Comment:
> will check that both the feature gate is enabled and the underlying OS is Linux

Yet another example of combining feature gate with other conditions. I think combining them is probably the right approach, but maybe the capability name should be something different if it's not strictly just reporting the feature gate status?